pkg/transforms/mqtt.go

```diff
@@ -134,25 +136,36 @@ func (sender MQTTSender) MQTTSend(edgexcontext *appcontext.Context, params ...in
 		// We didn't receive a result
 		return false, errors.New("No Data Received")
 	}
+
+	exportData, err := util.CoerceType(params[0])
+	if err != nil {
+		return false, err
+	}
+
 	if !sender.client.IsConnected() {
 		edgexcontext.LoggingClient.Info("Connecting to mqtt server")
 		if token := sender.client.Connect(); token.Wait() && token.Error() != nil {
+			sender.setRetryData(edgexcontext, exportData)
 			return false, fmt.Errorf("Could not connect to mqtt server, drop event. Error: %s", token.Error().Error())
```

Comment:
Not sure about this log message saying drop event anymore...should reflect if it was persisted or not

Reply:
Good catch.